ui/raidboss/popup-text.ts

```diff
@@ -350,30 +536,30 @@ export class PopupText {
     this.triggers = [];
     this.netTriggers = [];
     let timelineFiles = [];
-    let timelines = [];
-    const replacements = [];
+    let timelines: string[] = [];
+    const replacements: TimelineReplacement[] = [];
     const timelineStyles = [];
     this.resetWhenOutOfCombat = true;
 
     const orderedTriggers = new OrderedTriggerList();
 
     // Recursively/iteratively process timeline entries for triggers.
     // Functions get called with data, arrays get iterated, strings get appended.
-    const addTimeline = (function(obj) {
+    const addTimeline = (function(this: PopupText, obj: TimelineFunc) {
       if (Array.isArray(obj)) {
-        for (let i = 0; i < obj.length; ++i)
-          addTimeline(obj[i]);
+        for (const objVal of obj)
+          addTimeline(objVal);
       } else if (typeof obj === 'function') {
         addTimeline(obj(this.data));
       } else if (obj) {
         timelines.push(obj);
       }
     }).bind(this);
 
-    // construct something like regexEn or regexFr.
+    // construct something like regexDe or regexFr.
     const langSuffix = this.parserLang.charAt(0).toUpperCase() + this.parserLang.slice(1);
-    const regexParserLang = 'regex' + langSuffix;
-    const netRegexParserLang = 'netRegex' + langSuffix;
+    const regexParserLang = ('regex' + langSuffix) as RegexParserLang;
+    const netRegexParserLang = ('netRegex' + langSuffix) as NetRegexParserLang;
```

Comment:
I think you could remove both of these `as` type assertions (and the types themselves) if you changed the code using them below to:

```diff
@@ -629,9 +626,10 @@ export class PopupText {
           let found = false;

           // parser-language-based regex takes precedence.
+          const triggerObject: { [key: string]: unknown } = trigger;
           if (isRegexTrigger(trigger)) {
-            const regex: RegExp | undefined = trigger[regexParserLang] ?? trigger.regex;
-            if (regex) {
+            const regex = triggerObject[regexParserLang] ?? trigger.regex;
+            if (regex instanceof RegExp) {
               trigger.localRegex = Regexes.parse(regex);
               orderedTriggers.push(trigger);
               found = true;
@@ -639,8 +637,8 @@ export class PopupText {
           }

           if (isNetRegexTrigger(trigger)) {
-            const netRegex = trigger[netRegexParserLang] || trigger.netRegex;
-            if (netRegex) {
+            const netRegex = triggerObject[netRegexParserLang] ?? trigger.netRegex;
+            if (netRegex instanceof RegExp) {
               trigger.localNetRegex = Regexes.parse(netRegex);
               orderedTriggers.push(trigger);
               found = true;
```

(Sorry for attached diff.  Github really doesn't like suggestions that extend over old code.)